runtime/src/accounts_background_service.rs

```diff
@@ -646,7 +674,12 @@ impl AccountsBackgroundService {
                                 .flush_accounts_cache(force_flush, Some(max_clean_slot_inclusive));
 
                             if should_clean {
-                                bank.clean_accounts();
+                                bank.rc.accounts.accounts_db.clean_accounts(
```

Comment:
To be safe, maybe we can assert max_clean_slot_inclusive is less or eqal to bank.slot()?